test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs

```diff
@@ -89,6 +57,7 @@ protected virtual void AddAdditionalServices(IServiceCollection services)
             // points to the root folder of the test project.
             // To compensate, we need to calculate the correct project path and override the application
             // environment value so that components like the view engine work properly in the context of the test.
+            var startupAssembly = typeof(TStartup).GetTypeInfo().Assembly;
             var applicationName = startupAssembly.GetName().Name;
             var library = libraryManager.GetLibrary(applicationName);
             var applicationRoot = Path.GetDirectoryName(library.Path);
```

Comment:
Do you need to reset the `hostingEnvironment` still? Doesn't it rebase itself on the new app base?